File: py/src/text/generate.py
import boto3
import json
import pprint

client = boto3.client(service_name='bedrock-runtime', region_name="us-west-2")

titan_model_id = 'amazon.titan-text-express-v1'

titan_config = json.dumps({
            "inputText": "Tell me a story about a dragon",
            "textGenerationConfig": {
                "maxTokenCount": 4096,
                "stopSequences": [],
                "temperature": 0,
                "topP": 1
            }
        })

llama_model_id = "meta.llama3-70b-instruct-v1:0"
llama_config = json.dumps({
    "prompt": "Tell me a story about a dragon",
    "max_gen_len": 512,
    "temperature": 0,
    "top_p": 0.9,
})

response = client.invoke_model(
    body=llama_config,
    modelId=llama_model_id,
    accept="application/json",
    contentType="application/json"
)

response_body = json.loads(response.get('body').read())

pp = pprint.PrettyPrinter(depth=4)
# pp.pprint(response_body.get('results')) # titan config
pp.pprint(response_body["generation"]) # llama config

File: ts/src/text/generate.ts
import { BedrockRuntimeClient, InvokeModelCommand } from '@aws-sdk/client-bedrock-runtime'

const client = new BedrockRuntimeClient({ region: 'us-west-2' })

const titanConfig = {
    inputText: 'Tell me a story about a dragon.',
    textGenerationConfig : {
        maxTokenCount: 4096,
        stopSequences: [],
        temperature: 0,
        topP: 1,
    },
};

const titanModelId = 'amazon.titan-text-express-v1';

const llamaConfig = {
    prompt: 'Tell me a story about a dragon.',
    max_gen_len: 512,
    temperature: 0,
    top_p: 0.9,
}
const llamaModelId = "meta.llama3-70b-instruct-v1:0"


async function invokeModel(){

    const response = await client.send(new InvokeModelCommand({
        body: JSON.stringify(llamaConfig),
        modelId: llamaModelId,
        contentType: "application/json",
        accept: "application/json",
    }));

    const responseBody = JSON.parse(new TextDecoder().decode(response.body))
    // console.log(responseBody) // titan
    console.log(responseBody.generation) // llama
}

invokeModel()